etc/kayobe/ansible/filter_plugins/filters.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python3
+
+from ansible.errors import AnsibleFilterError
+
+class FilterModule(object):
+    def filters(self):
+        return {
+            'group_hostvars_by_var':
+            self.group_hostvars_by_var,
+            'group_hostvars_by_host':
+            self.group_hostvars_by_host
+        }
+
+    def group_hostvars_by_var(self, hostvars, var, stdout=None):
+        """
+        Returns a dictionary where the keys are values for the
+        specified var in hostvars, and the values are the hosts
+        that match that value.
+        """
+        result = {}
+
+        for host in hostvars.keys():
+            try:
+                indiv_host_all = hostvars[host]
+                indiv_host_var = indiv_host_all[var]
+                if stdout is not None:
+                    indiv_host_var = indiv_host_var[stdout]
+                result.setdefault(indiv_host_var, []).append(host)
+            except KeyError as e:
+                raise AnsibleFilterError(f"Variable {var} not found for host {host} in hostvars: {e}")
+
+        return result
+
+    def group_hostvars_by_host(self, hostvars, var, stdout=None, hostgroup=None):
+        result = {}
+        """
+        Returns a dictionary where the keys are hosts and the values
+        are the values for the specified var in hostvars. If hosts is
+        specified, only those hosts are included.
+        """
+        for host in hostvars.keys():
+            try:
+                indiv_host_vars = hostvars[host][var]
```

Comment:
You're selecting a single variable here, so probably best to use the singular
```suggestion
                indiv_host_var = hostvars[host][var]
```